graylog2-server/src/main/java/org/graylog/security/shares/EntitySharesService.java

```diff
@@ -258,6 +264,23 @@ public EntityShareResponse updateEntityShares(GRN ownedEntity, EntityShareReques
                 .build();
     }
 
+    /**
+     * Applies the share request to dependent entities, that we want to keep in sync.
+     *
+     * @param ownedEntity the parent entity
+     * @param shareRequest the sharing request to apply to the dependent entities
+     * @param sharingUser the sharing user
+     * @return list of updated dependent entities
+     */
+    private Set<GRN> resolveImplicitGrants(GRN ownedEntity, EntityShareRequest shareRequest, User sharingUser) {
+        Set<GRN> dependentEntities = entitiesResolvers.stream()
+                .flatMap(resolver -> resolver.dependentEntities(ownedEntity).stream())
+                .collect(Collectors.toSet());
+
+        dependentEntities.forEach(grn -> updateEntitySharesWithoutDependencies(grn, shareRequest, sharingUser));
```

Comment:
What should happen if the `validateRequest` call in `updateEntitySharesWithoutDependencies` fails? (can it fail in practice?) Right now, the method returns, leaving us with only one side of the share sync done. We don't do the synced entity resolving in `prepareShare`, so we don't get a validation error before the user submits the share action.

Reply:
It doesn't fail, if the entities are in sync. If they fall out of sync, validation may fail.
For now, I just added logging. Feels a bit too heavy-handed to abort the entire entity creation or update, just because we couldn't apply permissions to a related entity. Users had to manage their permissions entirely manually in the past, so they should be able to get things back in sync, if something was inadvertently changed in isolation.